packages/types/src/redirects.ts

```diff
@@ -0,0 +1,34 @@
+import type { OAuthStrategy, SamlStrategy } from './strategies';
+
+export type AuthenticateWithRedirectParams = {
+  /**
+   * Full URL or path to the route that will complete the OAuth flow.
+   * Typically, this will be a simple `/sso-callback` route that calls `Clerk.handleRedirectCallback`
+   * or mounts the <AuthenticateWithRedirectCallback /> component.
+   */
+  redirectUrl: string;
+
+  /**
+   * Full URL or path to navigate after the OAuth flow completes.
+   */
+  redirectUrlComplete: string;
+
+  /**
+   * Whether to continue (i.e. PATCH) an existing SignUp (if present) or create a new SignUp.
+   */
+  continueSignUp?: boolean;
+
+  /**
+   * One of the supported OAuth providers you can use to authenticate with, eg 'oauth_google'.
+   * Or alternatively `saml`, to authenticate with SAML.
+   */
+  strategy: OAuthStrategy | SamlStrategy;
```

Comment:
🔧 Let's add more type safety here with conditional types. When strategy equals `SamlStrategy` the `identifier` key will be required. All the other keys, are available for both strategies.

Reply:
Discussed this with @nikosdouvlis.

While discriminated union types are nice for strictness, they make consumption e.g. in `authenticateWithRedirect` very tricky, because props can no longer easily be destructured.

Reply:
This makes sense to me, its a discussion of type strictness versus ease of use.
@yourtallness can you verify that FAPI returns an appropriate/readable error if `identnfier` is not passed in?
Also, if you have any examples from our discussion that demonstrate the issues caused by the extra strictness, could you please share them for future reference?

Reply:
@nikosdouvlis Checked that `identifier` & `email_address` respectively are not submitted when not provided, so we should be OK.

Reply:
Will get back to you with examples with regard to discriminated union types.